### PR TITLE
add a new python script topology.py and its corresponding unittest fo…

### PR DESCRIPTION
**Pull request by wuziyou199217**
### PR types
New features

### PR changes
Others

### Describe
This PR add a new python script topology.py and its corresponding unittest for single-node topology detection

### CODE REVIEW

**Review by hitywt — APPROVED** [review]
LGTM

**Review by aoyulong — APPROVED** [review]
LGTM

File: python/paddle/distributed/auto_parallel/topology.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+            link_bandwidths.pop(i * self.nb_devices + i)
+
+        cmd = "rm matrix.txt"
+        err_msg = "Failed to delete matrix.txt"
+        default_value = ""
+        call_cmd(cmd, err_msg, default_value)
+
+        for i in range(len(link_types)):
+            link = {}
+            link['source_global_id'] = link_source_global_ids[i]
```
Comment (aoyulong) [inline python/paddle/distributed/auto_parallel/topology.py:327]: Are the `global_id` actually the `local_id`?

**Review by aoyulong — APPROVED** [review]

File: python/paddle/distributed/auto_parallel/topology.py (hunk 1 above)
Comment (wuziyou199217) [reply python/paddle/distributed/auto_parallel/topology.py:327]: in single-node case, yep

**Pull request merged**